src/_gettsim/interface.py

```diff
@@ -430,8 +436,13 @@ def _fail_if_group_variables_not_constant_within_groups(data):
         Dictionary containing a series for each column.
 
     """
+    hard_coded_groupings = [
+        key
+        for key, value in SUPPORTED_GROUPINGS.items()
```

Comment:
I think this check is conceptually slightly off. It should check also check those columns that I pass. E.g., if I pass `bg_id` instead of having GETTSIM calculate it, it should flag wrong values.